File: server/src/repository/shrt.ts
import { Shrt } from "../models/shrtTable"

export class ShrtRepository {
    async getUrlBySlug(slug: string): Promise<Shrt | null> {
        const link: Shrt | null = await Shrt.findOne({
            where: {
                slug: slug
            }
        })
        return link
    }

    async updateVisit(slug: string) {
        const link = await Shrt.increment({
            visit: 1
        }, {
            where: {
                slug: slug
            }
        })
        return link
    }
}

File: server/repository/shrt.go
package repository

import (
	"shrt-server/types/entity"
)

type ShrtRepository interface {
	FindBySlug(slug string) (*entity.Shrt, error)
	FindByLongURL(longURL string) (*entity.Shrt, error)
	UpdateVisit(id uint) error
	Create(shrt *entity.Shrt) error
}
